event_model/documents/generate/typeddict_to_schema.py

```diff
@@ -33,13 +31,28 @@
 from event_model.documents.generate.type_wrapper import (
     ALLOWED_ANNOTATION_ELEMENTS,
     AsRef,
+    BaseConfig,
+    BaseModel,
+    Field,
+    FieldInfo,
+    create_model,
     extra_schema,
+    pydantic_version,
 )
 
 # The hacky indexing on types isn't possible with python < 3.9
 if sys.version_info[:2] < (3, 9):
     raise EnvironmentError("schema generation requires python 3.8 or higher")
 
+if not pydantic_version:
+    raise EnvironmentError(
+        "schema generation requires pydantic < 2.0 to run, pydantic isn't installed"
+    )
+elif int(pydantic_version.split(".")[0]) >= 2:
+    raise EnvironmentError(
+        "schema generation requires pydantic < 2.0 to run, current version "
+        f"{pydantic_version}"
+    )
```

Comment:
Do we need this now if we're doing the v1 thing?

Reply:
Actually, I'll leave in the not pydantic_version. I think it's beneficial to have in case someone tries to touch the generation code from an environment without pip, or from event-model without dev requirements.